### ISSUE TITLE
Improve audio recording

### ISSUE THREAD

**Issue by TheMystery28** [xournalpp/xournalpp#4437]
- Record audio and link them to a specific (or several) page.
- When we press the option (or shortcut) the audio play. (for me the audio is saved but doesn't play on Windows)
- Better save audio file management (sorted and organized, named by folder so we don't have to do it. or just a way to custom how it is done)

That's all.

Thanks for the software.

**Comment by lunakid** [comment]
Yes, please! (For one, the voice memo support ~~is~~ would be the main reason ~~I use~~ I would use Xournal. Actually, I've just downloaded (for the first time) the latest (nightly Win) build to play with that feature for a try.)

A few specific notes (phrased more-or-less as requirements; hopefully some are already there (I just couldn't find them via the UI), or I'm banging on open doors; also, I'm kinda pretending here that things like OLE (and similar solutions for other canvas-oriented UIs/platforms) haven't been invented for >30 years :) Also, I'm blurring the "audio recording" concept with the more general "audio item" here, which may need refining later (e.g. per workflow considerations), as the feature matures):

**[audio-obj]** Audio recordings (i.e. the files themselves) should be represented as (and/or linked to) objects (as there can be any number of them for a given page/doc; there's also a "Play Object" command already -- which, BTW, I couldn't get to play back any recording yet). 
I'll refer to the "objectified" audio recordings as "audio objects" below.
("Objectifying" audio recordings would allow e.g. a) proper contextualization of multiple voice memos of a page, b) manipulating them in general (like any other "media file objects"), c) adding metadata to the raw audio files; etc.)

**[create-audio-obj-ui]** Existing recordings (actually any audio files anywhere) should be possible to browse and add as new audio objects, either directly to the canvas, or to an off-canvas palette (e.g. for drag-and-dropping; see [link-to-*], and also [add-unlinked].)

**[1-obj-1-file]** One audio object represents one audio file.

**[audio-obj-not-file]** The audio object, however, is not the same concept as the audio (recording) file. 
(E.g. If the file is not available (might just be offline), the audio object is still there all right, it just can't play. Or -> [audio-data-embed].)

**[audio-obj-not-link]** The audio object is also not just a "shortcut", but a proper first-class media object (despite only holding the filename to a recording).

**[audio-obj-immutable]** Audio objects are immutable (i.e. their content, the audio recording, is considered "final" (mostly to avoid potential inconsistencies regarding [link-to-obj-*]; see later).
_(TBD: a newly started recording may need to immediately create an audio object, before finishing the recording, for UX reasons (e.g. the user may need to manipulate it before an hour-long recording finishes...), violating either this "immutable" constraint or the "construction is initialization" principle.)_

**[audio-obj-edit-meta]** ...but their metadata can be edited, most importantly the audio file path/URL.
_(Note: [audio-obj-not-link] => the path is not the "content". See also [*-embed].)_

**[many-obj-1-file]** One audio file can be represented by multiple audio objects. (This may make little practical sense, but explicitly allowing it (rather than leaving it undefined or trying to forbid it) could remove a lot of headache.)

**[audio-data-embed]** Audio objects can even have no associated file at all, but contain (embed) audio data directly (in whatever suitable way).

**[link-to-obj]** An audio object should be possible to link to non-audio object(s). (E.g. by simple drag-and-drop.)
**[link-to-many-obj]** ...to any number of them.
**[link-many-to-1-obj]**? Linking multiple audio objects to one non-audio object should be fine, too.
**[typed-link-to-obj]**? If object links are typed in Xournal, then such audio links should be the same relations as any other annotations attached to some part fo the doc. (IOW: there could be other type of audio (and other media) links, too, but that's out-of-scope here now.)

**[link-to-last-obj]** If there's a "last/focused object" concept in Xournal, the audio obj. of a newly started recording should be linked to that by default.
If there isn't such an obj. state, then a _selected_ object should be the default anchor object. 

**[link-to-last-multi]** If multiple objects are selected, it should link to each. 
(That kinda follows from [link-to-many-obj], but it may better be explicit. Also, depending on [audio-obj-link]: if a selected obj. already has linked audio, the new recording should be either added to it, or should replace the previously linked one.)

**[add-unlinked]** If no object is "focused" (or selected) when starting a recording, then a new "raw" (unassociated) audio object should be added to the canvas (to provide a default way to manipulate them). (Similarly how [create-audio-obj-ui] would do it.)

**[audio-marker]** If an object has been linked to an audio object, it should have a corresponding visual clue on the UI (e.g. a little speaker overlay decoration, optionally (and also by default?) hover-only).

**[play-non-audio-obj]** Applying the "Play" command on a non-audio object should implicitly play a(ny) linked audio objects. (Naturally, when users (double-)click on an audio marker, that should trigger Play on the given object.)

**[remove-link]** The association of an audio object to some other object should be possible to remove easily (as supposedly any other kinds of object links, if there are such things in Xournal). (E.g. by right-clicking the [audio-marker] decoration and selecting "Remove" from a context menu popup...)

**[unlinked-obj-ui]** It should be possible to browse all audio _objects_ (as opposed to audio files; cf. [create-audio-obj-ui]) created "in the context" (i.e. "while"?...) editing a doc., regardless of those recordings being linked or not to any objects in that doc. (Note: audio links could be removed by the user (-> [remove-link]), also potentially removing all obvious UI clues to existing audio items -- but those audio objects still must be accessible to further manipulation somehow.)

**Comment by vext01** [comment]
I agree that the audio features are not intuitive.

(e.g. see https://github.com/xournalpp/xournalpp/issues/4779)
